### PR TITLE
gh-127415: add `PyMem_Raw{New,Resize}` convenience macros

### PR DESCRIPTION
**Pull request by picnixz**
* Issue: gh-127415

----
📚 Documentation preview 📚: https://cpython-previews--127416.org.readthedocs.build/

### CODE REVIEW

File: Include/pymem.h [hunk 1]
```diff
@@ -60,20 +60,21 @@ PyAPI_FUNC(void) PyMem_Free(void *ptr);
...
+ * The value of (PTR) is always clobbered by this macro regardless of success.
+ * The caller MUST check if (PTR) is NULL afterwards and deal with the memory
+ * error if so.  This means the original value of (PTR) MUST be saved for the
  * caller's memory error handler to not lose track of it.
  */
-#define PyMem_Resize(p, type, n) \
-  ( (p) = ((size_t)(n) > PY_SSIZE_T_MAX / sizeof(type)) ? NULL :        \
-        (type *) PyMem_Realloc((p), (n) * sizeof(type)) )
-
+#define PyMem_Resize(PTR, TYPE, N)                          \
```
Comment (rruuaanng) [inline Include/pymem.h:74]: ```PyMem_Resize(p, type, n)```
Would it be more appropriate to use lowercase, since `PyMem_Del` uses lowercase?

Comment (picnixz) [reply Include/pymem.h:74]: PEP-7 recommends having uppercase parameter names. I decided to cleanup the code because the file is quite small.

Comment (rruuaanng) [reply Include/pymem.h:74]: Oh， yes, I almost forgot :(

**Comment by picnixz** [comment]
Closing as `not planned` per https://github.com/python/cpython/issues/127415#issuecomment-2521152971.

**Pull request closed without merging**